velox/vector/DictionaryVector.h

```diff
@@ -131,9 +131,7 @@ class DictionaryVector : public SimpleVector<T> {
   }
 
   BufferPtr mutableIndices(vector_size_t size) {
-    // TODO: change this to isMutable(). See
-    // https://github.com/facebookincubator/velox/issues/6562.
-    if (indices_ && !indices_->isView() &&
+    if (indices_ && indices_->isMutable() &&
```

Comment:
Can we use BaseVector::resizeIndices here similar to how we use it in mutableOffsets and mutableSizes?